src/tup/create_name_file.c

```diff
@@ -80,6 +80,17 @@ int make_dirs_normal(struct tup_entry *dtent)
 		tup_db_del_ghost_tree(dtent);
 		dtent = dtent->parent;
 	}
+	if(dtent) {
+		/* Mark normal parent's .gitignore as
+		* needing update (t9010)
+		*/
+		struct tup_entry *gitignore_tent;
+		if(tup_db_select_tent(dtent, ".gitignore", &gitignore_tent) < 0)
+			return -1;
+		if(gitignore_tent && gitignore_tent->type == TUP_NODE_GENERATED) {
+			tup_db_add_create_list(gitignore_tent->dt); // HACK -- ideally, tup_db_add_modify_list here would be sufficient, but as it is, we need to reparse the directory's Tupfile to update the .gitignore.
```

Comment:
The create_list is a bit misnamed these days. Originally "The UPdater" code (aka: tup) was supposed to take a list of files that were created, deleted, or modified since the last update from the file monitor. These became the create_list, delete_list, and modify_list. Tupfiles needed to be re-parsed when files were created or deleted (or if the Tupfile itself was modified). Over time the delete_list was removed, and the create_list became a list of directories that need to be re-parsed rather than a list of newly created files. So adding the directory tupid to the create_list is the correct thing to do here.

Reply:
Just to confirm: I should switch this to be `dtent->dt` instead of `gitignore_tent->dt`? (:

Reply:
Sorry for the confusion -- `dt` is the tupid for the directory containing the file, so `gitignore_tent->dt` is correct. The tupid for the gitignore file itself would be `gitignore_tent->tnode.tupid` (the awkward construction there is because all of the tupids are in a tree in src/tup/entry.c). So `dtent->tnode.tupid` would equal `gitignore_tent->dt` here. `dtent->dt` is the parent directory of the directory containing the .gitignore file, so if we had `foo/sub/.gitignore`, then `dtent->dt` would be the directory `foo`, while we want `sub`.

Now it's probably more confusing :)

Reply:
Ah... okay, yeah, that makes sense. ~~Gotta love unintentionally being right; I for sure thought I was adding the gitignore as a created file there XDXD~~